django_project/changes/tests/test_views.py

```diff
@@ -752,6 +752,369 @@ def test_VersionDelete_no_login(self):
         self.assertEqual(response.status_code, 302)
 
 
+class TestVersionViewsWithAnonymouseUserForCRUD(TestCase):
+    """
+    Check if anonymous user can perform CRUD operations on version entries
+    just in case they have the URL to the views.
+    """
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def setUp(self):
+        """
+        Setup before each test
+        We force the locale to en otherwise it will use
+        the locale of the host running the tests and we
+        will get unpredictable results / 404s
+        """
+
+        self.client = Client()
+        self.client.post(
+            '/set_language/', data={'language': 'en'})
+        logging.disable(logging.CRITICAL)
+        self.project = ProjectF.create(name='testproject')
+        self.version = VersionF.create(
+            project=self.project,
+            name='1.0.1')
+        self.category = CategoryF.create(
+            project=self.project,
+            name='testcategory')
+
+        self.user = None
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def tearDown(self):
+        """
+        Teardown after each test.
+
+        :return:
+        """
+        self.project.delete()
+        self.version.delete()
+        self.category.delete()
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_VersionListView_with_anonymouse_user(self):
```

Comment:
Please search and replace "mouse" for "mouse"

Reply:
anonymous

Reply:
Updated @timlinux